src/main/java/com/fundynamic/d2tm/game/map/Cell.java

```diff
@@ -66,7 +59,7 @@ public Entity getEntity() {
 
     public void setEntity(Entity entity) {
         if (this.entity != null) {
-            throw new IllegalArgumentException("Cannot place mapEntity (" + entity + ") because mapEntity already exists: " + this.entity);
+            throw new CellAlreadyOccupiedException("Cannot place Entity (" + entity + ") on cell because Entity already present: " + this.entity);
```

Comment:
the whole concept of Cells having a pointer to an Entity will be gone soon. Perhaps when doing infantry, perhaps even sooner. But this will not survive long. :)

Short answer: no this will not work for infantry :)